Update Pinyin library and small adjustments

ElectroHeavenVN opened the pull request:
# Pull Request Title
Update Pinyin library and small adjustments
## Description:
This pull request includes several changes to update the `pinyin` library to `pinyin-pro` and adjust related code accordingly. Additionally, it includes modifications to the `parseLyrics` function and its tests, as well as some minor text adjustments.
## Changelog:

- ### 🎉 New Features and Updates

- ### 🔨 Fixes and Improvements

- ### 🐜 Known Issues and Bugs

## Other Issues:

## Checklist:

- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Sandakan/Nora/pulls) for the same update/change?
- [x] Have you linted your code locally before submission with `npm lint` and `npm run prettier-write` commands?
- [x] Have you successfully run the app with your changes locally with `npm run dev`?

Sandakan commented:
@ElectroHeavenVN,
There are some issues with the packages that you imported.
![image](https://github.com/user-attachments/assets/8a78c0a3-8bca-431d-be1e-ff90a962a758)

You may have not noticed them because I incorrectly asked to try the `npm lint` instead of the `npm run lint` command to check the whole workspace for errors.
Can you fix the issues in the files displayed when the `npm run lint` command is run?
![image](https://github.com/user-attachments/assets/9526d741-5fb8-4196-8421-a286f7612044)

ElectroHeavenVN commented:
> @ElectroHeavenVN, There are some issues with the packages that you imported. ![image](https://private-user-images.githubusercontent.com/45460443/394803059-8a78c0a3-8bca-431d-be1e-ff90a962a758.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5ODY5MzAsIm5iZiI6MTczMzk4NjYzMCwicGF0aCI6Ii80NTQ2MDQ0My8zOTQ4MDMwNTktOGE3OGMwYTMtOGJjYS00MzFkLWJlMWUtZmY5MGE5NjJhNzU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDA2NTcxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlMjJiMzQ0ZGFlZTc2OTkyODJlZTZjZTkyYTA1NjY0YjQ2MmY4MjBiY2U0MzdmYTUzNzAxZTVlOGFlODFlYmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.IxbWqL7LbNTmNapSgMAURsY9tWNzajC_Vfj1Lm95gsA)
> 
> You may have not noticed them because I incorrectly asked to try the `npm lint` instead of the `npm run lint` command to check the whole workspace for errors. Can you fix the issues in the files displayed when the `npm run lint` command is run? ![image](https://private-user-images.githubusercontent.com/45460443/394804033-9526d741-5fb8-4196-8421-a286f7612044.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5ODY5MzAsIm5iZiI6MTczMzk4NjYzMCwicGF0aCI6Ii80NTQ2MDQ0My8zOTQ4MDQwMzMtOTUyNmQ3NDEtNWZiOC00MTk2LTg0MjEtYTI4NmY3NjEyMDQ0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDA2NTcxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUwYzJhMTBkZjcwNmYwNDI4ZDBmZGYxZmU1MWNhMzdkY2Q2Nzg3OWYxZGFhZWVlNmYyOGE4ZjIwNjg2NDVhMmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._cmy0tMWr_MTenekQqU-WUyKFJTyTQaiAeWmVHuLdSg)



> @ElectroHeavenVN, Always try to use packages with full typescript support so that you don't have to manually generate them.
> 
> https://www.npmjs.com/package/@sglkc/kuroshiro https://www.npmjs.com/package/@sglkc/kuroshiro-analyzer-kuromoji

I fixed errors and warnings stated in the `npm run lint` command and updated the libraries. You can review them when you have time.

ElectroHeavenVN commented:
I created a new one here: #293
